src/db_adapters/lists.py

```diff
@@ -0,0 +1,70 @@
+from typing import Literal, overload
+
+from sqlalchemy.ext.asyncio import AsyncSession
+
+from src.db_adapters.media import ensure_media
+from src.db_tables.user_list import UserList, UserListItem, UserListItemKind
+
+
+@overload
+async def list_put_item(
+    session: AsyncSession,
+    user_list: UserList,
+    tvdb_id: int,
+    kind: Literal[UserListItemKind.MOVIE, UserListItemKind.SERIES],
+) -> UserListItem: ...
+
+
+@overload
+async def list_put_item(
+    session: AsyncSession,
+    user_list: UserList,
+    tvdb_id: int,
+    kind: Literal[UserListItemKind.EPISODE],
+    series_id: int,
+) -> UserListItem: ...
+
+
+async def list_put_item(
+    session: AsyncSession, user_list: UserList, tvdb_id: int, kind: UserListItemKind, series_id: int | None = None
+) -> UserListItem:
+    """Add an item to a user list, raises an error if the item is already present."""
+    await ensure_media(session, tvdb_id, kind, series_id=series_id)
+    item = UserListItem(list_id=user_list.id, tvdb_id=tvdb_id, kind=kind)
+    session.add(item)
+    await session.commit()
+    return item
+
+
+async def list_get_item(
+    session: AsyncSession, user_list: UserList, tvdb_id: int, kind: UserListItemKind
+) -> UserListItem | None:
+    """Get an item from a user list."""
+    return await session.get(UserListItem, (user_list.id, tvdb_id, kind))
+
+
+async def list_remove_item(session: AsyncSession, user_list: UserList, item: UserListItem) -> None:
```

Comment:
Do we even need to take the `user_list` here? Shouldn't every item be aware of what list it belongs to?

Reply:
It's for utility, so that you don't have to refresh the list.